src/DynamoCore/Core/UndoRedoRecorder.cs

```diff
@@ -241,6 +244,22 @@ public XmlElement PopFromUndoGroup()
             return PopActionGroupFromUndoStack();
         }
 
+        /// <summary>
+        /// A model is recorded as an off-track object means it is modified
+        /// somewhere else, but that modification operation is not recorded in
+        /// undo/redo stack. UndoRedoRecorder will ignore all excpetions that
+        /// related to this kind of objects during undo/redo.
+        /// 
+        /// For example, a connector that connects to an input port of a custom
+        /// node instance could be deleted because of the removal of that input
+        /// port in custom workspace. As this deletion is not recorded by
+        /// UndoRedoRecorder, the connector should be marked as off-track.
```

Comment:
Perhaps a better to describe this last statement would be:

`As this deletion in the custom workspace is not recorded by UndoRedoRecorder in home workspace, the connector should be marked as off-track`

Also IMHO, the method should then be renamed to `MarkModelAsOffTrack` since it does not involve any sort of recording (just to avoid confusion).